packages/jest-snapshot/src/getMatchers.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+'use strict';
+
+const common = require('./common');
+const fs = require('fs');
+const path = require('path');
+const serializer = require('./serializer');
+const Snapshot = require('./Snapshot');
+
+const testsByPath =  {};
+const getNextIndexFor = (test, filePath) => {
+  const tests = testsByPath[filePath] || (testsByPath[filePath] = {});
+  if (test in tests) {
+    return ++tests[test];
+  }
+  return tests[test] = 0;
+};
+
+module.exports = (filePath, options) => ({
+  toMatchSnapshot: (util, customEquality) => {
+    return {
+      compare(actual, expected) {
+        // Retrieving last test asap to avoid race condition
+        const lastTest = common.getLastTest(filePath);
+
+        const snapshotsPath = path.join(path.dirname(filePath), '__snapshots__');
+        try {
+          const folder = fs.statSync(snapshotsPath);
+          if (!folder.isDirectory()) {
+            fs.mkdirSync(snapshotsPath);
+          }
+        } catch (e) {
+          fs.mkdirSync(snapshotsPath);
+        }
+        // the extension has to be not a jest test one.
+        const snapshotsName = path.join(
```

Comment:
let's just call it `snapshotFileName`? With `snapshotsName` it's not entirely clear what it is about.

I also recommend doing this: `const SNAPSHOT_EXTENSION = '.snap';` on the top of the file and then use it here. That way it is clear what we are doing and you don't need the comment.